Userland/Applications/Piano/ExportProgressWindow.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ExportProgressWindow.h"
+#include "LibGUI/Icon.h"
+#include <AK/DeprecatedString.h>
+#include <Applications/Piano/ExportProgressWidget.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
+
+ExportProgressWindow::ExportProgressWindow(GUI::Window& parent_window, Atomic<int>& wav_percent_written)
+    : GUI::Dialog(&parent_window)
+    , m_wav_percent_written(wav_percent_written)
+{
+    auto& main_widget = set_main_widget<GUI::Widget>();
+    if (!main_widget.load_from_gml(export_progress_widget))
+        VERIFY_NOT_REACHED();
+
+    set_resizable(false);
+    set_closeable(false);
+    set_title("Rendering audio");
+    set_icon(GUI::Icon::default_icon("app-piano"sv).bitmap_for_size(16));
+
+    m_progress_bar = *main_widget.find_descendant_of_type_named<GUI::HorizontalProgressbar>("progress_bar");
+    m_label = *main_widget.find_descendant_of_type_named<GUI::Label>("export_message");
+
+    start_timer(250);
+}
+
+void ExportProgressWindow::set_filename(StringView filename)
+{
+    m_label->set_text(DeprecatedString::formatted("Rendering audio to {}…", filename));
+    update();
+}
+
+void ExportProgressWindow::timer_event(Core::TimerEvent&)
+{
+    m_progress_bar->set_value(m_wav_percent_written.load());
+    if (window_id() != 0)
+        set_progress(m_wav_percent_written.load());
+
+    if (m_wav_percent_written == 100)
```

Comment:
Set the percentage back to 0 here? Or elsewhere. It looks like otherwise, we can't save a file more than once with the same Piano instance.